homeassistant/components/homeassistant_sky_connect/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""The Home Assistant Sky Connect integration."""
+from __future__ import annotations
+
+from homeassistant.components import usb
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a Home Assistant Sky Connect config entry."""
+    usb_info = usb.UsbServiceInfo(
+        device=entry.data["device"],
+        vid=entry.data["vid"],
+        pid=entry.data["pid"],
+        serial_number=entry.data["serial_number"],
+        manufacturer=entry.data["manufacturer"],
+        description=entry.data["description"],
+    )
+    if not usb.async_is_plugged_in(hass, entry.data):
+        # The USB dongle is not plugged in
+        raise ConfigEntryNotReady
```

Comment:
This is a bit meh. Maybe the `usb` integration should fire events when a device is plugged or unplugged?

Reply:
The observer only works on some of the supported platforms, and if the host has a new enough udev.  There is fallback to scan the ports when you load the integrations page

Reply:
Can you elaborate a bit on for which install types it works and for which it doesn't?

Reply:
@bdraco should work on OS, we can call websocket API or retrigger an scan if needed on unplugged hardware